app.js

```diff
@@ -235,30 +235,46 @@
                 message: message
             });
         } else {
-            User.register({
-                username: req.body.username
-            }, req.body.password, function(err, user) {
-                if (err) {
-                    console.log(err.message);
-                    var message = err.message;
-                    res.render("register", {
-                        message: message
-                    });
-                } else {
-                    User.updateOne({
-                        _id: user._id
-                    }, {
-                        email: req.body.email,
-                        fname: req.body.name,
-                        isAdmin: false
-                    }, function(err) {
-                        if (err) console.log(err);
-                    });
-                    passport.authenticate("local")(req, res, function() {
-                        res.redirect("/UserHome");
-                    });
-                }
-            });
+            if(req.body.username.indexOf(' ') >= 0) {
+                var message = "username should not contain whitespace";
+                res.render('register', {
```

Comment:
it should be `return res.render(...)`
Otherwise controll will flow to rest of the function also and cause error.

I tested & it is throwing error.